### ISSUE TITLE
Add touchpad swipe gesture support in the Flutter Embedder

### ISSUE THREAD

**Issue by whiskeyPeak** [flutter/flutter#152943]
### Use case

Currently the trackpad APIs only allow for 2-finger scrolling type gestures. This means that flutter applications and embedders have no way to introduce any kind of 3-finger swipe gestures to their users. A large majority of trackpads also support that feature. This change would allow for 3 finger or more gestures on the trackpad, dramatically improving Flutter's gesture abilities.

### Proposal

I propose we add an API that either simply produces events for 3 swipe gestures or 4 swipe gestures, or an API where we report back the number of fingers on the trackpad and let flutter decide the internal logic.

**Comment by moffatman** [comment]
We could add pointerCount to PointerPanZoom? Right now I just assume it to be 2 everywhere

**Comment by whiskeyPeak** [comment]
Hmm yeah that would be the cleanest solution. I can't speak for MacOs and Windows but for GTK3, we aren't exposed to the number of fingers on the trackpad directly so we'll need to manually add 3 finger and 4 finger events.

**Comment by whiskeyPeak** [comment]
I don't have the permission to assign this to myself so fyi I'm working on the Linux side of things :)

**Comment by jonahwilliams** [comment]
Doesn't the embedder already support this? Does the framework need support for these gestures?

**Comment by jonahwilliams** [comment]
@chinmaygarde double checked, the engine supports this. @jason-simmons mentions that maybe the linux embedder is missing this?

**Comment by whiskeyPeak** [comment]
What APi refers to the pointer counter in the Embedder.h file?

**Comment by moffatman** [comment]
I think a bit of confusion in terms here. Seems like the only actual work is to add pointer count. But there is some mixup with "swipe" vs "scroll"

**Comment by whiskeyPeak** [comment]
Ah yes, I'm referring to the idea of being able to scroll with 2 fingers or more. So a PointerPanZoomEvent with a pointer count.

**Comment by moffatman** [comment]
I didn't do it in the first place since on every other platform, 3/4 finger gestures are reserved for operating system gestures like window management

**Comment by whiskeyPeak** [comment]
Yes but this becomes an issue when flutter is used in Embedded devices or as part of a Desktop Environment.

**Comment by moffatman** [comment]
No problem with adding it just explaining why it's not there yet. Although the more you go towards a full DE or something, you may need to change the API shape, to expose individual finger touch positions. But from what you posted so far I don't think it's necessary.
